src/NuGet.Core/NuGet.Configuration/Settings/SettingBase.cs

```diff
@@ -8,11 +8,11 @@ namespace NuGet.Configuration
 {
     public abstract class SettingBase
     {
-        internal XNode Node { get; private set; }
+        internal XNode? Node { get; private set; }
 
-        internal ISettingsGroup Parent { get; set; }
+        internal ISettingsGroup? Parent { get; set; }
 
-        internal SettingsFile Origin { get; private set; }
+        internal SettingsFile? Origin { get; private set; }
```

Comment:
A lot of code assumes this property is never null, and will throw NullReferenceException if it is. Yet, many classes set this to null in `Update` methods.

In the end it was too hard to figure out an ideal fix, so I annotated the property as it's used (nullable), and suppressed the warning in a few places where NullReferenceException will be thrown (unless there was an obvious/easy way to handle the null).  This PR is trying to add null annotations, not redesign the class design.